parcels/baseparticlefile.py

```diff
@@ -0,0 +1,326 @@
+    def getuid():
+        return 'tmp'
+
+
+__all__ = ['BaseParticleFile']
+
+
+def _set_calendar(origin_calendar):
+    if origin_calendar == 'np_datetime64':
+        return 'standard'
+    else:
+        return origin_calendar
+
+
+class BaseParticleFile(ABC):
+    """Initialise trajectory output.
+
+    :param name: Basename of the output file
+    :param particleset: ParticleSet to output
+    :param outputdt: Interval which dictates the update frequency of file output
+                     while ParticleFile is given as an argument of ParticleSet.execute()
+                     It is either a timedelta object or a positive double.
+    :param write_ondelete: Boolean to write particle data only when they are deleted. Default is False
+    :param convert_at_end: Boolean to convert npy files to netcdf at end of run. Default is True
+    :param tempwritedir: directories to write temporary files to during executing.
+                     Default is out-XXXXXX where Xs are random capitals. Files for individual
+                     processors are written to subdirectories 0, 1, 2 etc under tempwritedir
+    :param pset_info: dictionary of info on the ParticleSet, stored in tempwritedir/XX/pset_info.npy,
+                     used to create NetCDF file from npy-files.
+    """
+
+    def __init__(self, name, particleset, outputdt=np.infty, write_ondelete=False, convert_at_end=True,
+                 tempwritedir=None, pset_info=None):
+
+        self.write_ondelete = write_ondelete
+        self.convert_at_end = convert_at_end
+        self.outputdt = outputdt
+        self.lasttime_written = None  # variable to check if time has been written already
+
+        self.dataset = None
+        self.metadata = {}
+        if pset_info is not None:
+            for v in pset_info.keys():
+                setattr(self, v, pset_info[v])
+        else:
+            self.name = name
+            self.particleset = particleset
+            self.parcels_mesh = 'spherical'
+            if self.particleset.fieldset is not None:
+                self.parcels_mesh = self.particleset.fieldset.gridset.grids[0].mesh
+            self.time_origin = self.particleset.time_origin
+            self.lonlatdepth_dtype = self.particleset.collection.lonlatdepth_dtype
+            self.var_names = []
+            self.var_names_once = []
+            for v in self.particleset.collection.ptype.variables:
+                if v.to_write == 'once':
+                    self.var_names_once += [v.name]
+                elif v.to_write is True:
+                    self.var_names += [v.name]
+            if len(self.var_names_once) > 0:
+                self.written_once = []
+                self.file_list_once = []
+
+            self.file_list = []
+            self.time_written = []
+            # ==== distinction between ID-based and index-based write done in derived class ==== #
+            # self.maxid_written = -1
```

Comment:
This commented line can go?

Reply:
yes, it can - gonna take care of it today